### PR TITLE
Implement the pluggable scanner framework - stage 3

### LINKED ISSUE #8985: REST API implementations for the interrogation service (pluggable scanner)

### PR DESCRIPTION
**Pull request by steven-zou**
Implement the pluggable scanner framework - stage 3

Signed-off-by: Steven Zou <szou@vmware.com>

- add scan resource and update role bindings
- update registration model and related interfaces
- implement scan API to do scan/get report/get log
- update repository rest API to produce scan report summary
- update scan job hook handler
- update some UT cases
- update robot account making content
- hidden credential in the job log
- Comment scan related API test cases which will be re-activate later

fix #8985

### CODE REVIEW

File: make/migrations/postgresql/0012_1.10.0_schema.up.sql [hunk 1]
```diff
@@ -23,8 +23,9 @@ CREATE TABLE scan_report
     digest VARCHAR(256) NOT NULL,
     registration_uuid VARCHAR(64) NOT NULL,
     mime_type VARCHAR(256) NOT NULL,
-    job_id VARCHAR(32),
-    status VARCHAR(16) NOT NULL,
+    job_id VARCHAR(64),
+    track_id VARCHAR(64),
+    status VARCHAR(1024) NOT NULL,
```
Comment (ywk253100) [inline make/migrations/postgresql/0012_1.10.0_schema.up.sql]: There are two migration sql files for 1.10: 0011 and 0015, how about merging them into one?

Comment (steven-zou) [reply make/migrations/postgresql/0012_1.10.0_schema.up.sql]: Agree, we can do it in a new PR

File: src/core/api/pro_scanner.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+// GetProjectScanner gets the project level scanner
+func (sa *ProjectScannerAPI) GetProjectScanner() {
+	// Check access permissions
+	resource := rbac.NewProjectNamespace(sa.pid).Resource(rbac.ResourceConfiguration)
+	if !sa.SecurityCtx.Can(rbac.ActionRead, resource) {
+		sa.SendForbiddenError(errors.New(sa.SecurityCtx.GetUsername()))
+		return
+	}
```
Comment (heww) [inline src/core/api/pro_scanner.go]: There is a method `RequireProjectAccess` in `BaseController`, that can do permission checking simpler.

```go
if !sa.RequireProjectAccess(sa.pid, rbac.ActionRead, rbac.ResourceConfiguration) {
        return
}
```

File: src/core/api/pro_scanner.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+// SetProjectScanner sets the project level scanner
+func (sa *ProjectScannerAPI) SetProjectScanner() {
+	// Check access permissions
+	resource := rbac.NewProjectNamespace(sa.pid).Resource(rbac.ResourceConfiguration)
+	if !sa.SecurityCtx.Can(rbac.ActionUpdate, resource) {
+		sa.SendForbiddenError(errors.New(sa.SecurityCtx.GetUsername()))
+		return
+	}
```
Comment (heww) [inline src/core/api/pro_scanner.go]: There is a method RequireProjectAccess in BaseController, that can do permission checking simpler.

File: src/core/api/scan.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+// Scan artifact
+func (sa *ScanAPI) Scan() {
+	// Check access permissions
+	resource := rbac.NewProjectNamespace(sa.pro.ProjectID).Resource(rbac.ResourceScan)
+	if !sa.SecurityCtx.Can(rbac.ActionCreate, resource) {
+		sa.SendForbiddenError(errors.New(sa.SecurityCtx.GetUsername()))
+		return
+	}
```
Comment (heww) [inline src/core/api/scan.go]: There is a method RequireProjectAccess in BaseController, that can do permission checking simpler.

File: src/core/api/scan.go [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+// Report returns the required reports with the given mime types.
+func (sa *ScanAPI) Report() {
+	// Check access permissions
+	resource := rbac.NewProjectNamespace(sa.pro.ProjectID).Resource(rbac.ResourceScan)
+	if !sa.SecurityCtx.Can(rbac.ActionRead, resource) {
+		sa.SendForbiddenError(errors.New(sa.SecurityCtx.GetUsername()))
+		return
+	}
```
Comment (heww) [inline src/core/api/scan.go]: There is a method RequireProjectAccess in BaseController, that can do permission checking simpler.

File: src/core/api/scan.go [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+// Log returns the log stream
+func (sa *ScanAPI) Log() {
+	// Check access permissions
+	resource := rbac.NewProjectNamespace(sa.pro.ProjectID).Resource(rbac.ResourceScan)
+	if !sa.SecurityCtx.Can(rbac.ActionRead, resource) {
+		sa.SendForbiddenError(errors.New(sa.SecurityCtx.GetUsername()))
+		return
+	}
```
Comment (heww) [inline src/core/api/scan.go]: There is a method RequireProjectAccess in BaseController, that can do permission checking simpler.

File: src/core/api/pro_scanner.go [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+	// Get ID of the project
+	pid, err := sa.GetInt64FromPath(":pid")
+	if err != nil {
+		sa.SendBadRequestError(errors.Wrap(err, "scanner API: get project scanners"))
+		return
+	}
+	sa.pid = pid
```
[the remark this answers is not in the document]
Comment (steven-zou) [reply src/core/api/pro_scanner.go]: ok, as a project verification.

**Pull request merged**